skupstina_django/scraper/scrape_utils_html.py

```diff
@@ -0,0 +1,172 @@
+
+    acts = []
+    for i, act_url in enumerate(act_urls):
+        site = requests_retry_session().get(root_url + act_url).content
+        soup = BeautifulSoup(site, 'html.parser')
+        act_content = get_visible_text(soup)
+        act_title = act_titles[i]
+        acts.append(
+            {
+                'act_content': act_content,
+                'act_url': act_url,
+                'act_title': act_title,
+                'act_file_type': 'HTML',
+            }
+        )
+
+    # Check for word or pdf attachments
+    if "<a href='" in text:
+        # Regex to extract link to document
+        docu_urls = re.findall("<a href='(.*)','Dokument", text)
+        for docu_url in docu_urls:
+            docu_title, docu_raw_data, docu_file_type = parse_document_link(docu_url)
+            acts.append(
+                {
+                    'act_content': docu_raw_data,
+                    'act_url': docu_url,
+                    'act_title': docu_title,
+                    'act_file_type': docu_file_type,
+                }
+            )
+    subject_details['acts'] = acts
+    return subject_details
+
+
+def scrape_engine(act_period: str, periods_url: str) -> None:
+    """
+    Scraper engine used to scrape open act.
+
+    :param str act_period:
+        Example: '20. siječnja 2020. - 24.siječnja 2020'
+
+    :param str periods_url:
+        Example: 'http://web.zagreb.hr/sjednice/2017/Sjednice_2017.nsf/DRJ?OpenAgent&'
+    """
+    act_period = act_period.strip().lower()
+    print('\nScraping period: ', act_period)
+    url = (periods_url + act_period).replace(' ', '%20').lower()
+    print(url)
+    period_obj = write_period_to_db(act_period, url)
+    subjects, num_els = parse_subjects_list(url)
+    for subject in subjects:
+        parse_complete = False
+        max_retries = 10
+        sleep_time = 10
+        print('Parsing ', subject['subject_url'])
+        while not parse_complete and max_retries > 0:
+            try:
+                subject_details = parse_subject_details(subject['subject_url'])
+                parse_complete = True
+            except exceptions.ConnectionError as e:
+                parse_complete = False
+                max_retries -= 1
+                print('Connection Error while parsing {}:\n{}\n'.format(subject['subject_url'], e))
+                print('Retrying...\n')
+                time.sleep(sleep_time)
+        if max_retries == 0:
+            print('Maximum retries exceeded. Please run the scraper again.\n')
+            raise exceptions.ConnectionError
+        subject['details'] = subject_details
+        subject_obj = write_subject_to_db(subject, period_obj)
+        write_act_to_db(subject['details']['acts'], subject_obj)
+        print('Parsed  ', subject['subject_url'], ' **')
+
+
+def scrape_last(last_n_periods: int, url_suffix: str) -> None:
+    """
+    Scrapes the last n periods once again from the scrapes_completed file.
```

Comment:
Nice that you started adding docstrings. Those are really useful now as the codebase is increasing in size.